### PR TITLE
FW-580. Port iot-lab_M3 board with IAR

### PR DESCRIPTION
**Pull request by keomabrun**
https://openwsn.atlassian.net/browse/FW-580
The following projects were ported with IAR:
* bsp_timer
* bsp_leds
* bsp_radio_tx
* bsp_radio_timer
* bsp_uart
* drv_opentimers
* oos_mercator
* oos_openwsn

The projects compile without warning but have not been tested with experimentation.

IAR version: 6.60.2 (for ARM)

### CODE REVIEW

File: bsp/boards/iot-lab_M3/leds.c [hunk 1]
```diff
@@ -136,7 +136,9 @@ uint8_t leds_radio_isOn()
 void leds_debug_on() {}
 void leds_debug_off() {}
 void leds_debug_toggle() {}
-uint8_t leds_debug_isOn() {}
+uint8_t leds_debug_isOn() {
+  return 0;
```
Comment (changtengfei) [inline bsp/boards/iot-lab_M3/leds.c]: Shouldn't just return 0

File: bsp/boards/iot-lab_M3/spi.c [hunk 2]
```diff
@@ -43,14 +43,14 @@ volatile spi_vars_t spi_vars;
 inline static void RESET_CLR(void) { GPIOC->BRR = 1<<1; }
 inline static void RESET_SET(void) { GPIOC->BSRR = 1<<1; }
 inline static void CSn_SET(void) { GPIOA->BSRR = 1<<4; }
-inline static void CSn_CLR(void) { GPIOA->BRR = 1<<4; }
+//inline static void CSn_CLR(void) { GPIOA->BRR = 1<<4; } //never referenced
```
Comment (changtengfei) [inline bsp/boards/iot-lab_M3/spi.c]: remove the line if it's not useful at all?

**Comment by changtengfei** [comment]
@keomabrun  just left two comments on this PR, once it's solved I will merge it :-)

File: bsp/boards/iot-lab_M3/spi.c (hunk 2 above)
Comment (keomabrun) [inline bsp/boards/iot-lab_M3/spi.c]: Got inspired by: https://github.com/openwsn-berkeley/openwsn-fw/blob/develop/bsp/boards/agilefox/spi.c#L47
Should I remove it then ?

File: bsp/boards/iot-lab_M3/leds.c (hunk 1 above)
Comment (changtengfei) [inline bsp/boards/iot-lab_M3/leds.c]: I see, maybe add a comment to indicate that like the link you shown.

Comment (keomabrun) [inline bsp/boards/iot-lab_M3/leds.c]: Done. https://github.com/openwsn-berkeley/openwsn-fw/pull/313/commits/7642763f90f7cee279e19ad15420a352fe8c2c3d
and 8d174e6

**Comment by keomabrun** [comment]
Note that the projects are only compiling. There have not been tested on the motes yet.

File: bsp/boards/iot-lab_M3/library/STM32F10x_StdPeriph_Lib_V3.5.0/Libraries/STM32F10x_StdPeriph_Driver/src/stm32f10x_rtc.c [hunk 3]
```diff
@@ -336,24 +336,4 @@ void RTC_ClearITPendingBit(uint16_t RTC_IT)
   * @}
   */
 
-/******************* (C) COPYRIGHT 2011 STMicroelectronics *****END OF FILE****/
```
Comment (twatteyne) [inline bsp/boards/iot-lab_M3/library/STM32F10x_StdPeriph_Lib_V3.5.0/Libraries/STM32F10x_StdPeriph_Driver/src/stm32f10x_rtc.c]: I'm a bit surprised about this change. What does this have to do with IAR? Can you confirm things still compile using `armgcc`?

Comment (twatteyne) [inline bsp/boards/iot-lab_M3/library/STM32F10x_StdPeriph_Lib_V3.5.0/Libraries/STM32F10x_StdPeriph_Driver/src/stm32f10x_rtc.c]: OK!

**Comment by changtengfei** [comment]
@keomabrun is that possible that I compile the project with iar toolchain?

**Comment by keomabrun** [comment]
I've never tried using scons
